packages/dom-expressions/src/client.js

```diff
@@ -460,7 +460,6 @@ function insertExpression(parent, value, current, marker, unwrapArray) {
   if (value === current) return current;
   const t = typeof value,
     multi = marker !== undefined;
-  parent = (multi && current[0] && current[0].parentNode) || parent;
```

Comment:
Not sure what might break by this removal but this for sure breaks the parent comparison in `cleanChildren`

Reply:
Ryan made a comment about this code in a related issue here https://github.com/ryansolid/dom-expressions/pull/307#issuecomment-1904967433 
I will have to study it again to refresh my mind, I'm not sure I actually ever understood that part

Reply:
I've tried to track this down and, oh well 😅, no wonder even Ryan can't remember this. This check existed since the beginning.